src/compiler/emitter.ts

```diff
@@ -760,7 +760,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 
             function getGeneratedNameForNode(node: Node) {
                 const id = getNodeId(node);
-                return nodeToGeneratedName[id] || (nodeToGeneratedName[id] = unescapeIdentifier(generateNameForNode(node)));
+                return getOrUpdateArray(nodeToGeneratedName, id, () => unescapeIdentifier(generateNameForNode(node)));

```

Comment:
This introduces a new closure and environment record for every call to `getGeneratedNameForNode`. The original approach does not.